File: chap07/run-model7-1.R
library(cmdstanr)

d <- read.csv(file='input/data-rental.csv')
Np <- 50
Xp <- seq(from=10, to=120, length=Np)
data <- list(N=nrow(d), X=d$Area, Y=d$Y, Np=Np, Xp=Xp)

model <- cmdstan_model(stan_file='model/model7-1.stan')
fit <- model$sample(data=data, seed=123, parallel_chains=4)
fit$save_object(file='output/result-model7-1.RDS')

File: chap07/run-model7-1.py
import pandas
import numpy as np
import cmdstanpy

d = pandas.read_csv('input/data-rental.csv')
Np = 50
Xp = np.linspace(10, 120, Np)
data = {'N':len(d), 'X':d.Area, 'Y':d.Y, 'Np':Np, 'Xp':Xp}

model = cmdstanpy.CmdStanModel(stan_file='model/model7-1.stan')
fit = model.sample(data=data, seed=123, parallel_chains=4)
fit.save_csvfiles('output/result-model7-1')
